src/fields/fields.service.ts

```
@@ -303,7 +305,114 @@ export class FieldsService {
+      if (!obj || typeof obj !== 'object') {
+        return null;
+      }
+
+      // If this object has a fieldId that matches, return it
+      if (obj.fieldId === fieldId) {
+        return obj;
+      }
+
+      // Recursively search through all object values
+      for (const key in obj) {
+        if (obj.hasOwnProperty(key)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)